src/parser/mod.rs

```diff
@@ -11173,15 +11173,42 @@ impl<'a> Parser<'a> {
                     self.peek_token().location
                 )
             }
-            expr => self
-                .parse_optional_alias(keywords::RESERVED_FOR_COLUMN_ALIAS)
-                .map(|alias| match alias {
-                    Some(alias) => SelectItem::ExprWithAlias { expr, alias },
-                    None => SelectItem::UnnamedExpr(expr),
-                }),
+            expr => {
+                if dialect_of!(self is MsSqlDialect) {
+                    if let Some(select_item) = self.parse_mssql_alias_with_equal(&expr) {
+                        return Ok(select_item);
+                    }
+                }
+                self.parse_optional_alias(keywords::RESERVED_FOR_COLUMN_ALIAS)
+                    .map(|alias| match alias {
+                        Some(alias) => SelectItem::ExprWithAlias { expr, alias },
+                        None => SelectItem::UnnamedExpr(expr),
+                    })
+            }
         }
     }
 
+    /// Parse a [`SelectItem`] based on an MsSql syntax that uses the equal sign
+    /// to denote an alias, for example: SELECT col_alias = col FROM tbl
+    /// <https://learn.microsoft.com/en-us/sql/t-sql/queries/select-examples-transact-sql?view=sql-server-ver16#b-use-select-with-column-headings-and-calculations>    
+    fn parse_mssql_alias_with_equal(&mut self, expr: &Expr) -> Option<SelectItem> {
```

Comment:
```suggestion
    fn maybe_unpack_alias_assignment(expr: &Expr) -> Option<SelectItem> {
```
Thinking since this is only a helper function that it doesn't get confused for a parser/parsing method